src/Sarif/Core/Run.cs

```diff
@@ -13,6 +14,109 @@ public partial class Run
         private static Invocation EmptyInvocation = new Invocation();
         private static LogicalLocation EmptyLogicalLocation = new LogicalLocation();
 
+        private IDictionary<FileLocation, int> _fileToIndexMap;
+
+        public int GetFileIndex(
+            FileLocation fileLocation, 
+            bool addToFilesTableIfNotPresent = true,
+            OptionallyEmittedData dataToInsert = OptionallyEmittedData.None,
+            System.Text.Encoding encoding = null)
```

Comment:
>System.Text [](start = 12, length = 11)

Just curious how you decide when to inline this and when to add a `using`. I suppose your criterion is "only used once." And maybe that's better than my practice of always adding a `using`. It fits with the goal of reducing cognitive load ("Where is `Encoding` defined?")

Reply:
My personal rule: inlined fully qualified type name is permissible when used a single time or when required to resolve a type name collision. A using statement is fine as well (except in cases when it requires creating an alias, in which case I generally prefer the fully qualified name).

Somewhere, there is a collision with the 'Encoding' term. I assumed I'd provoke it here but actually did not. Reverted to 'using.

---
In reply to: [244521675](https://github.com/Microsoft/sarif-sdk/pull/1191#discussion_r244521675) [](ancestors = 244521675)